Sources/Fuzzilli/Core/CodeGenerators.swift

```diff
@@ -421,46 +421,33 @@ public let CodeGenerators: [CodeGenerator] = [
 
     CodeGenerator("DestructObjectGenerator", input: .object()) { b, obj in
         var properties = Set<String>()
-        for _ in 0...5 {
-            if let prop = b.type(of: obj).properties.randomElement(), !properties.contains(prop){
+        for _ in 0..<Int.random(in: 2...6) {
+            if let prop = b.type(of: obj).properties.randomElement(), !properties.contains(prop) {
                 properties.insert(prop)
+            } else {
+                properties.insert(b.genPropertyNameForRead())
             }
         }
 
-        // If we haven't added any properties, add a random property name
-        if properties.isEmpty {
-            properties.insert(b.genPropertyNameForRead())
-        }
-
         let hasRestElement = probability(0.2)
 
-        // Ensure that we have at least one output
-        guard !properties.isEmpty || hasRestElement else { return }
-
         b.destruct(obj, selecting: properties.sorted(), hasRestElement: hasRestElement)
     },
 
     CodeGenerator("DestructObjectAndReassignGenerator", input: .object()) { b, obj in
         var candidates: [Variable] = []
         var properties = Set<String>()
-        for _ in 0...5 {
-            if let prop = b.type(of: obj).properties.randomElement(), !properties.contains(prop){
+        for _ in 0..<Int.random(in: 2...6) {
+            if let prop = b.type(of: obj).properties.randomElement(), !properties.contains(prop) {
                 properties.insert(prop)
                 candidates.append(b.randVar())
+            } else {
+                properties.insert(b.genPropertyNameForRead())
+                candidates.append(b.randVar())
```

Comment:
I think here it could happen that the `insert` doesn't actually insert anything (because the property is already in there), in which case you'd end up with more `candidates` than `properties`.
Maybe it's easier to just compute the candidates separately, after the properties. E.g. with something like

```swift
var variables = properties.map { b.randVar() }
```
or just a regular for-loop

Reply:
yeah good catch, I'll add the changes